pkg/worker/workflow.go

```diff
@@ -161,11 +167,57 @@ func (w *worker) TriggerModelActivity(ctx context.Context, param *TriggerModelAc
 	// 	return nil, w.toApplicationError(err, param.ModelID, ModelActivityError)
 	// }
 
+	modelTags := make([]string, 0)
+	for _, tag := range param.ModelTags {
+		modelTags = append(modelTags, tag.TagName)
+	}
+	jsonBytes, _ := json.Marshal(modelTags)
+
+	start := time.Now()
+
+	runLog, err := w.repository.CreateModelTrigger(ctx, &datamodel.ModelTrigger{
```

Comment:
nit: we should split `TriggerModelActivity` into several activities. By adding more and more actions to this activity we're missing the point of durable executions.